web3/providers/ipc.py

```diff
@@ -37,18 +47,18 @@ def get_ipc_socket(ipc_path, timeout=0.1):
 class PersistantSocket:
     sock = None
 
-    def __init__(self, ipc_path):
+    def __init__(self, ipc_path: str) -> None:
         self.ipc_path = ipc_path
 
-    def __enter__(self):
+    def __enter__(self) -> socket.socket:
         if not self.ipc_path:
             raise FileNotFoundError("cannot connect to IPC socket at path: %r" % self.ipc_path)
 
         if not self.sock:
             self.sock = self._open()
         return self.sock
 
-    def __exit__(self, exc_type, exc_value, traceback):
+    def __exit__(self, exc_type: Type[BaseException], exc_value: Any, traceback: Any) -> None:
```

Comment:
`exc_value` should be `BaseException`.  `traceback` should be `types.TracebackType`